hera_pspec/uvwindow.py

```diff
@@ -75,152 +78,153 @@ def __init__(self, pol, spw_range=None, ftfile=None,
 
-        Returns
+        Given the path to a beam simulation, obtain the Fourier transform 
+        of the instrument beam in the sky plane for all the frequencies 
+        in the spectral window. 
+        Output is an array of dimensions (kperpx, kperpy, freq).
+        Computations correspond to the Fourier transform performed in equation 
+        10 of memo.
+
+        Parameters
         ----------
-        bandwidth : array of floats
-            List of frequencies covered by the instrument, in Hz.
+        beamfile : str
+            Path to file containing beam simulation/
+        verbose : bool, optional
+            If True, print progress, warnings and debugging info to stdout.
+        x_orientation: str, optional
+            Orientation in cardinal direction east or north of X dipole.
+            Default keeps polarization in X and Y basis.
+            Used to convert polstr to polnum and conversely.
         """
-        assert self.ft_file is not None,\
-            "Can only be used if self.ft_file is defined."
+        raise NotImplementedError('Coming soon...')
 
-        filename = '{}_{}.hdf5'.format(self.ft_file, self.pol)
-        assert os.path.isfile(filename), \
-            "Cannot find FT beam file: {}.".format(filename)
-        with h5py.File(filename, "r") as f:
-            bandwidth = np.array(f['freq'][...])
+    @classmethod
+    def from_file(cls, ftfile, spw_range=None, verbose=False, x_orientation=None):
+        """
+        Read Fourier transform of beam in sky plane from file.
 
-        assert bandwidth.size > 1, "Error reading file, empty bandwidth."
+        Initialise FTBeam object by reading file containing the 
+        Fourier transform of the  instrument beam in the sky plane 
+        for given frequencies. 
 
-        return bandwidth
+        Parameters
+        ----------
+        ftfile : str
+            Path to file constraining to the Fourier transform of the 
+            beam on the sky plane (Eq. 10 in Memo), including the polarisation
+            Ex : path/to/file/ft_beam_HERA_dipole_pI.hdf5
+            File must be h5 format.
+        spw_range : tuple or 2-element array of ints
+            In (start_chan, end_chan). Must be between 0 and 1024 (HERA
+            bandwidth).
+            If None, whole instrument bandwidth is considered.
+        verbose : bool, optional
+            If True, print progress, warnings and debugging info to stdout.
+        x_orientation: str, optional
+            Orientation in cardinal direction east or north of X dipole.
+            Default keeps polarization in X and Y basis.
+            Used to convert polstr to polnum and conversely.
+        """
+        # check file path input
+        if not isinstance(ftfile, str):
+            raise TypeError('Wrong ftfile input. See docstring.')
+        if not os.path.isfile(ftfile):
+             raise ValueError("Cannot find FT beam file: {}.".format(ftfile))
```

Comment:
We should probably support the input `ftfile` being a `pathlib.Path` instance. This would allow you to change this code to

```python
from pathlib import Path
...
def from_file(cls, ftfile, ...):
    ftfile = Path(ftfile)  # if it's a string or Path this will work.
    if not (ftfile.exists() and ftfile.is_file()):
        raise ValueError(...)

Reply:
But if `Path` only takes a string or a path as input, won't I need a `isinstance(ftfile, str)` anyway to avoid that `Path(ftfile)` raises an error?

Reply:
Firstly, to the point of using it: it's becoming the standard way to deal with file paths in python, so it's not unlikely that someone will want to pass in a `Path` instance as the `ftfile`. They can always pass in `str(ftfile)` to convert it to string, but I think that covering both cases (since it's easy enough) is more future-proof.

Secondly, no you don't need to do the check. Doing `Path(ftfile)` will raise an error if `ftfile` is not str or Path, but it will be a nice understandable error that the user will be then able to fix. I.e. calling `Path(ftfile)` essentially *does* the type-checking for you.

Reply:
You convinced me!